### PR TITLE
Jetpack post preview

### LINKED ISSUE #5326: Posts: Unauthenticated webview when viewing a post from the posts list
### Expected behavior

I expect to get an authenticated webview when I select "View" on a published or draft post in the posts list.

### Actual behavior

The webview is not authenticated, so viewing a published post bumps the stats and viewing a draft leads to a 404 error.

h/t Davor for reporting this issue in the alpha version.

cc @maxme — possibly due to FluxC changes?

### Steps to reproduce the behavior

1. Open the Blog Posts section for a site.
2. Find a draft (marked "Draft" - not a local draft) in the posts list.
3. Tap "View" on the post's card. Result: 404 error.
4. Find a published post in the posts list.
5. Tap "View" on the post's card. Result: The post loads but with a logged-out view, so the view is counted in the site's stats.

##### Tested on Nexus 9, Android 7.1.1, WPAndroid alpha-38

### PR DESCRIPTION
**Pull request by oguzkocer**
Fixes #5326. This feature turned out to much simpler than what I understood it to be. Don't let the branch name fool you, we can't show authenticated version of a post without password for Jetpack sites. That was my misunderstanding.

This PR fixes the Jetpack blog posts not showing up at all in webview. We used to show a username/password error, because of treating a Jetpack site the same as self-hosted. We can only show the regular url of the post, and to handle previews we add a `frame-nonce` parameter to the url.

Relevant FluxC PR: https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/358

To test:
* Create a draft in a Jetpack site
* Tap on the "View" button in posts list, you should see the post even if it's a draft (remote one, of course)

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/WPWebViewActivity.java [hunk 1]
```diff
@@ -97,6 +97,16 @@ public static void openUrlByUsingGlobalWPCOMCredentials(Context context, String
         openWPCOMURL(context, url);
     }
 
+    public static void openJetpackBlogPostPreview(Context context, String url, String frameNonce) {
```
Comment (nbradbury) [inline WordPress/src/main/java/org/wordpress/android/ui/WPWebViewActivity.java]: Can we add a comment header here describing the purpose of `frameNonce`?

**Comment by nbradbury** [comment]
@oguzkocer Code looks good and appears to fix the problem. Once a comment header is added I'm good to :shipit:

File: WordPress/src/main/java/org/wordpress/android/ui/WPWebViewActivity.java (hunk 1 above)
Comment (oguzkocer) [inline WordPress/src/main/java/org/wordpress/android/ui/WPWebViewActivity.java]: Added in 6ded00f284fdb190475185a25d01daa30f3e07ed

**Comment by oguzkocer** [comment]
Thanks for the review @nbradbury, I actually had a comment for it, but it got lost in all the changes while I was figuring things out!

**Comment by nbradbury** [comment]
:shipit:

**Comment by maxme** [comment]
Should we merge this in `release/7.0`? Are preview broken for jetpack users currently in 7.0?

**Comment by oguzkocer** [comment]
@maxme it's up to you! If we are already using a new enough version of FluxC, I don't see why not as this PR is pretty simple.

**Comment by maxme** [comment]
@oguzkocer ok let's do this.

**Comment by oguzkocer** [comment]
@maxme I am guessing you wanted a PR instead of cherry picking, so here it is: https://github.com/wordpress-mobile/WordPress-Android/pull/5438

**Pull request merged**